jicoco-metrics/src/main/kotlin/org/jitsi/metrics/BooleanMetric.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright @ 2022 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.metrics
+
+import io.prometheus.client.Gauge
+
+/**
+ * A metric that represents booleans using Prometheus [Gauges][Gauge].
+ * A non-zero value corresponds to `true`, zero corresponds to `false`.
+ */
+class BooleanMetric @JvmOverloads constructor(
+    /** the name of this metric */
+    name: String,
+    /** the description of this metric */
+    help: String,
+    /** the namespace (prefix) of this metric */
+    namespace: String,
+    /** an optional initial value for this metric */
+    initialValue: Boolean = false
+) : Metric<Boolean> {
+    private val gauge =
+        Gauge.build(name, help).namespace(namespace).register().apply { set(if (initialValue) 1.0 else 0.0) }
```

Comment:
Can you use a `Gauge` without (or before) registering it? I think you may run into some awkwardness when adding tests if every `Metric` instance automatically registers itself with the default registry.

I think a better approach is to make registration the responsibility of MetricsContainer. This will also make it easy to allow the user of the library (this is now a library :)) to use their own registry (i.e. just add a `val` in MetricsContainer)

Reply:
> Can you use a `Gauge` without (or before) registering it? I think you may run into some awkwardness when adding tests if every `Metric` instance automatically registers itself with the default registry.

Yes, they can be used without being registered.

> I think a better approach is to make registration the responsibility of MetricsContainer. This will also make it easy to allow the user of the library (this is now a library :)) to use their own registry (i.e. just add a `val` in MetricsContainer)

I agree with separating registration from creation. I think we can streamline MetricsContainer. I'll get back to you on that.
I've also been thinking of replicating the builder pattern from the prometheus library for each metric type, it seems more intuitive.